ring/replication_set_tracker.go

```diff
@@ -63,8 +112,18 @@ func newZoneAwareResultTracker(instances []InstanceDesc, maxUnavailableZones int
 	for _, instance := range instances {
 		t.waitingByZone[instance.Zone]++
 	}
+
 	t.minSuccessfulZones = len(t.waitingByZone) - maxUnavailableZones
 
+	t.zoneContexts = make(map[string]context.Context, len(t.waitingByZone))
```

Comment:
These can be fairly large maps especially when running without zone-awareness. Also creating each `WithCancel` below allocates a go channel and a function on the heap. 

It wouldn't hurt to avoid them IMO. We can have a separate, say, `contextTracker` which is instantiated only in case of `DoUntilQuorum`.

Reply:
Makes sense, resolved in f467533